dist: ensure ts typedefs get shipped

Fang- opened the pull request:
Typescript type definitions (that already get generated during tsc compilation) were not being included in the package. We make two changes to resolve this:

First, we list `*.d.ts` files for inclusion in the package. The build process already generates these.

...Except for `src/types.d.ts`, which the compiler ignores because of its `.d.ts` extension (as opposed to plain `.ts`). Other type definition files still try referring to it (as `./types`) but that lookup fails because no `dist/types.*` file got created.
(Arguably it's a `tsc` bug that it creates files with broken imports, but we're not in the business of fixing those.)

So, secondly, we rename `types.d.ts` to `types.ts`, causing it to be picked up for processing by `tsc` and resulting in `dist/types.*` files.

Fang- committed:
dist: ensure ts typedefs get shipped

Typescript type definitions (that already get generated during tsc
compilation) were not being included in the package. We make two changes
to resolve this:

First, we list `*.d.ts` files for inclusion in the package. The build
process already generates these.

...Except for `src/types.d.ts`, which the compiler ignores because of
its `.d.ts` extension (as opposed to plain `.ts`). Other type definition
files still try referring to it (as `./types`) but that lookup fails
because no `dist/types.*` file got created.
(Arguably it's a `tsc` bug that it creates files with broken imports.)

So, secondly, we rename `types.d.ts` to `types.ts`, causing it to be
picked up for processing by `tsc` and resulting in `dist/types.*` files.

ntvangoor commented on src/types.ts:
I would prefer to keep the `.d.ts` extension as it should only contain type declarations. 
Could you find a way to fix the issue while keeping the extension?

Fang- replied:
From what I gather from microsoft/TypeScript#5112 and microsoft/TypeScript#35296 and its related issues, there's no way to do this with `tsc` natively.

I'm assuming that, per out-of-band discussion on #1, we still want to avoid extra build steps like copying/moving files around. Maybe there's some way to include the `src/*.d.ts` files and rejigger the imports/references such that they resolve cleanly. I'll poke around a bit more.

Fang- replied:
Yeah, you can write the imports in such a way that they resolve both from `src/` and from `dist/`. Please take another look!

Fang- committed:
dist: include .d.ts files from src/

Reverts the extension renaming done in 85a3d6b. Instead includes
`.d.ts` files from `src/` in the package files and updates the imports
referring to the `src/types.d.ts` to use a relative path instead.

(This works now because `src/` and `dist/` will contain files at the
same depth (& in the same structure), but may fail to resolve if that
ever changes.)

ntvangoor commented on src/load-saints.ts:
```diff
@@ -1,4 +1,4 @@
-import { LiturgicalClass } from "./types";
+import { LiturgicalClass } from "../src/types"; //NOTE intentional, must resolve from dist/
```
This is a confusing solution.
It would be best to copy the types file to the dist folder

Fang- replied:
I'm pretty sure we'd have to eat the tradeoff that we saw in #1 if we want to hard-copy the file, where:
> we are forced to add a dev dependency (`cpx`) [and additional build step] for copying the files over. Writing a simple `cp` or `rsync` would work on POSIX systems, but not Windows.

Are you ok with taking that tradeoff? Or is there another option here that you think I should try/think about first?

ntvangoor replied:
Yes, I guess that tradeoff is fine.

ntvangoor commented:
No longer applicable
